docs/prover/cli/conf-file-api.md

```diff
@@ -0,0 +1,143 @@
+Certora Prover Conf File Format
+===============================
+
+Conf files are an alternative way for setting arguments for the
+ `certoraRun` tool. In terms of functionality 
+using conf files is identical to the use of the [CLI Options](options.md). Instead of calling `certoraRun` 
+with a list of shell flags, some or all the flags can be stored in a JSON file 
+(to be more precise the format is [JSON5](https://json5.org/)):
+
+```
+certoraRun my_params.conf
+```
+
+* Conf files must use the `.conf` suffix
+
+
+
+How CLI Options are mapped to JSON
+----------------------------------
+
+Command-line arguments are stored as key-value pairs in the conf file. 
+The keys are the names of the parameters (with the leading `--` removed). 
+For example,
+```
+certoraRun --verify Example:example.spec
+```
+is equivalent to running with the following conf file:
+
+```
+{ "verify": "Example:example.spec" }
+```
+The values in the map depend on the type of arguments:
+
+* boolean flags that take no arguments (such as {ref}`--send_only`), 
+the value should be `true`. For example,
+```
+certoraRun --send_only
+```
+
+would be encoded as:
+```
+{ "send_only": true }
+```
+
+* flags that expect a single argument (such as {ref}`--solc`) or as {ref}`--loop_iter`) 
+ are encoded as a JSON string. For example,
+```
+certoraRun --solc solc4.25 --loop_iter 2
+```
+
+would be encoded as:
+```
+{ "solc": "solc4.25", "loop_iter": "2" }
+```
+
+Note that conf files do not use JSON numbers; numbers are encoded as strings.
+
+* flags that expect multiple arguments (such as {ref}`--packages`)
+are encoded as JSON lists. For example,
+```
+certoraRun --packages @balancer-labs/v2-solidity-utils=pkg/solidity-utils @balancer-labs/v2-vault=pkg/vault
+```
+would be encoded as:
+```
+{
+"packages": [
+    "@balancer-labs/v2-solidity-utils=pkg/solidity-utils",
+    "@balancer-labs/v2-vault=pkg/vault"
+    ]
+}
+```
+
+
+* The input files in the CLI API will be stored under the key **files**
+
+```
+certoraRun example.sol  ...
+```
+will appear in the conf file as:
+```
+{
+    ...
+    "files": "example.sol" 
+    ...
+}
+```
+
+**Map Value CLI Options**
+
+Flags in CLI API that are maps will be stored as **JSON Objects**. Example:
```

Comment:
How do I know whether a flag is a map?

Reply:
listed the 2 map flags --solc_map and --solc_optimize_map